source/src_pdiag/pdiag_double.cpp

```diff
@@ -1190,8 +1259,133 @@ void Pdiag_Double::diago_complex_begin(
+			delete[] htot;
+			delete[] stot;
+		}
+		delete[] vtot;
+
+		// convert wave function to band distribution
+			// and calculate the density matrix in the tranditional way
+			// redistribute eigenvectors to wfc / wfc_aug
+
+		ModuleBase::timer::tick("Diago_LCAO_Matrix","gath_eig_complex");
+        std::complex<double> *work=new std::complex<double>[maxnloc]; // work/buffer matrix
+        int naroc[2]; // maximum number of row or column
+        int info;
+        for(int iprow=0; iprow<dim0; ++iprow)
+        {
+            for(int ipcol=0; ipcol<dim1; ++ipcol)
+            {
+                const int coord[2]={iprow, ipcol};
+                int src_rank;
+                MPI_Cart_rank(comm_2D, coord, &src_rank);
+                if(myid==src_rank)
+                {
+                    BlasConnector::copy(nloc, wfc_2d.c, inc, work, inc);
+                    naroc[0]=nrow;
+                    naroc[1]=ncol;
+                }
+                info=MPI_Bcast(naroc, 2, MPI_INT, src_rank, comm_2D);
+                info=MPI_Bcast(work, maxnloc, MPI_DOUBLE_COMPLEX, src_rank, comm_2D);
+
+                if(this->out_lowf)
+                {
+                    std::complex<double> **ctot;
+                    if(myid==0)
+                    {
+                        ctot = new std::complex<double>*[GlobalV::NBANDS];
+                        for (int i=0; i<GlobalV::NBANDS; i++)
+                        {
+                            ctot[i] = new std::complex<double>[GlobalV::NLOCAL];
+                            ModuleBase::GlobalFunc::ZEROS(ctot[i], GlobalV::NLOCAL);
+                        }
+                        ModuleBase::Memory::record("Pdiag_Basic","ctot",GlobalV::NBANDS*GlobalV::NLOCAL,"cdouble");
+                    }
+					// mohan update 2021-02-12, delete BFIELD option
+					info=q2WFC_WFCAUG_CTOT_complex(myid, naroc, nb,
+							dim0, dim1, iprow, ipcol,
+							work, wfc, GlobalC::LOWF.WFC_K_aug[ik], ctot);
+                    std::stringstream ss;
+	                ss << GlobalV::global_out_dir << "LOWF_K_" << ik+1 << ".dat";
+                    // mohan add 2012-04-03, because we need the occupations for the
+                    // first iteration.
+                    WF_Local::write_lowf_complex( ss.str(), ctot, ik );//mohan add 2010-09-09
+                    if(myid==0)
+                    {
+                        for (int i=0; i<GlobalV::NBANDS; i++)
+                        {
+                            delete[] ctot[i];
+                        }
+                        delete[] ctot;
+                    }
+                }
+                else
+                {
+					// mohan update 2021-02-12, delte BFIELD option
+					info=q2WFC_WFCAUG_complex(naroc, nb,
+							dim0, dim1, iprow, ipcol,
+							work, wfc, GlobalC::LOWF.WFC_K_aug[ik]);
+				}
+            }
+        }
+        delete[] work;
+        ModuleBase::timer::tick("Diago_LCAO_Matrix","gath_eig_complex");
+	}
+	#endif
```

Comment:
Please add `//__CUSOLVER_LCAO` comment for `#endif`.

Reply:
Have added it.

Reply:
@caic99 The changes requested have been addressed. Would you please approve that?